statistics/selectivity.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package statistics
+
+import (
+	"math"
+
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/context"
+	"github.com/pingcap/tidb/expression"
+	"github.com/pingcap/tidb/mysql"
+	"github.com/pingcap/tidb/sessionctx/variable"
+	"github.com/pingcap/tidb/util/ranger"
+	"github.com/pingcap/tidb/util/types"
+)
+
+// If one condition can't be calculated, we will assume that the selectivity of this condition is 0.8.
+const selectionFactor = 0.8
+
+// calcBlock is used for calculating selectivity.
+type calcBlock struct {
```

Comment:
`calcBlock` is not clear. Maybe `exprSet` is better, which stands for a set of expressions.